include/deal.II/cgal/additional_data.h

```diff
@@ -0,0 +1,136 @@
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_cgal_additional_data_h
+#define dealii_cgal_additional_data_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_CGAL
+
+#  include <CGAL/boost/graph/Named_function_parameters.h>
+
+DEAL_II_NAMESPACE_OPEN
+namespace CGALWrappers
+{
+  /**
+   * Struct that must be used to pass additional
+   * arguments to the CGAL::Mesh_criteria_3 class (see
+   * https://doc.cgal.org/latest/Mesh_3/index.html for more information.)
+   *
+   * The arguments allow for fine control on the size, quality, and distribution
+   * of the cells of the final triangulation. CGAL uses Boost named parameters
+   * for these arguments in dimension three, i.e., they must be specified with
+   * the syntax `CGAL::parameters::parameter_name=parameter_value`, irrespective
+   * of their order. Accepted parameters are:
+   *
+   * - `CGAL::parameters::edge_size`: a constant
+   * providing a uniform upper bound for the lengths of
+   * curve edges. This parameter has to be set to a positive value when
+   * 1-dimensional features protection is used.
+   * - `CGAL::parameters::facet_angle`: a lower bound for the angles (in
+   * degrees) of the surface mesh facets.
+   * - `CGAL::parameters::facet_size`: a constant
+   * describing a uniform upper bound for the radii
+   * of the surface Delaunay balls.
+   * - `CGAL::parameters::facet_distance`: a constant
+   * describing a uniform upper bound for the distance
+   * between the facet circumcenter and the center of its surface Delaunay ball.
+   * - `CGAL::parameters::facet_topology`: the set of topological constraints
+   * which have to be verified by each surface facet. The default value is
+   * `CGAL::FACET_VERTICES_ON_SURFACE`. See CGAL::Mesh_facet_topology manual
+   * page to get all possible values.
+   * - `CGAL::parameters::cell_radius_edge_ratio`: an upper bound for the
+   * radius-edge ratio of the mesh tetrahedra.
+   * - `CGAL::parameters::cell_size`: a constant
+   * describing a uniform upper bound for the
+   * circumradii of the mesh tetrahedra.
+   *
+   * @note This struct must be instantiated with `dim=3`.
+   */
+  template <int dim>
+  struct AdditionalData
+  {
+    double facet_angle;
+    double facet_size;
+    double facet_distance;
+    double cell_radius_edge_ratio;
+    double cell_size;
+
+    AdditionalData(
+      double facet_a            = CGAL::parameters::is_default_parameter(true),
+      double facet_s            = CGAL::parameters::is_default_parameter(true),
+      double facet_d            = CGAL::parameters::is_default_parameter(true),
+      double cell_radius_edge_r = CGAL::parameters::is_default_parameter(true),
+      double cell_s             = CGAL::parameters::is_default_parameter(true))
```

Comment:
I double checked this and I don't understand why these are the default values. In particular, in my installed copy of `CGAL/boost/graph/Named_function_parameters.h` that function is defined as

```cpp
bool inline is_default_parameter(const internal_np::Param_not_found&)
{
  return true;
}

template <class T>
bool is_default_parameter(const T&)
{
  return false;
}
```
i.e., these are always `true` which will be converted to `1` - why do we do this?

Reply:
We also don't do this in 2D.

Reply:
This is what the CGAL::Mesh_criteria_3 uses as default values. We don't do this in 2d because in 2d we use Surface_mesh_criteria, which takes double values as input, and not named parameters.

Reply:
My original issue was that this `ignored` value (also written in CGAL documentation) is not directly something available in user codes. However, inside CGAL::Mesh_Criteria_3 they have:

https://github.com/CGAL/cgal/blob/a8e87526348d2d1faa2ea893747314853ee6868d/Mesh_3/include/CGAL/Mesh_criteria_3.h#L97-L111

which suggests to replace my `CGAL::parameters::is_default_parameter(true)` with their default (doubles) values, as we're allowing only doubles so far. All tests are still passing.

I've just rebased and force pushed this change.

Reply:
Looking at the lines you pointed out, I think we should replace the default values with std::limits<double>::max for the first guy, and zero for all the others.

Reply:
Sure, that one was added in one the next PRs. I'll add it here now.